src/pybind/mgr/dashboard/module.py

```diff
@@ -510,28 +510,36 @@ def serve(self):
         class Root(object):
             @cherrypy.expose
             def index(self):
-                active_uri = module.get_active_uri()
-                if active_uri:
-                    module.log.info("Redirecting to active '%s'", active_uri)
-                    raise cherrypy.HTTPRedirect(active_uri)
+                if Settings.STANDBY_REDIRECT_ENABLED:
```

Comment:
> I'd just a this new option to set the new HTTP code.

??? Don't understand that :-)

Reply:
Sorry, I wasn't exactly brilliant there.

I just re-read the ceph tracker issue, and tried to understand which use-case we are trying to deal with here:
- **Legacy behaviour**: ceph-dashboard with no proxy. Stand-by 3xx redirects to active dashboard. If for some reason that info is not available, it self-refreshes after 5 secs.
- **NEW: HAproxy**: while this could potentially require dashboard to be aware of the virtual service address (VIP:VPT):
  1. We barely use explicit service addresses: perhaps only the 3xx Redirect to the active mgr, and the Grafana API URL.
  1. The 3xx can be made customizable, so dashboard returns the HTTP status code required for the HAProxy to mark it as unavailable. For example, `503 - Service Unavailable` includes a [`Retry-After` header](https://tools.ietf.org/html/rfc2616#page-141) that might be aligned with HAProxy check interval.
  1. While I think the above fulfills the use-case, for any edge case a network admin could work around that with DNS aliases (e.g.: an internal DNS resolves `{x,y,z}.mgr` resolve to the internal IP addresses, while the external one aliases those to the Reverse Proxy address).

So, that said, rather than adding 2 new settings and 2 new possible states (404 and custom url), why not just adding a new one (e.g. `dashboard-standby-http-response: {300,400,500}` or `dashboard-standby-behaviour: {redirect,error}`), by default 301/307 (whichever the `Redirect` is), and it can be optionally set to 404, 5xx, etc.

For planning dashboard integrations to 3rd parties I think it's best to stick to the existing use-cases than to fulfill all possible (but not current) use-cases (e.g.: general reverse proxy).

Reply:
Done